fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AggregateStrategies.java

```diff
@@ -416,7 +416,9 @@ private List<PhysicalHashAggregate<Plan>> twoPhaseAggregateWithCountDistinctMult
                 .build();
         PhysicalHashAggregate<Plan> gatherLocalAgg = new PhysicalHashAggregate<>(
                 localAggGroupBy, localOutput, Optional.of(partitionExpressions),
-                new AggregateParam(AggPhase.LOCAL, AggMode.INPUT_TO_BUFFER),
+                new AggregateParam(AggPhase.LOCAL,
+                        logicalAgg.getGroupByExpressions().isEmpty() ? AggMode.INPUT_TO_RESULT
+                                : AggMode.INPUT_TO_BUFFER),
```

Comment:
core with `select count(distinct k1, NULL), avg(k2) from test;`
maybe we set AggMode to INPUT_TO_RESULT, if and only if have no other agg function and have no group by exprs.

Reply:
or, it must be three stage?